aten/src/ATen/SparseTensorImpl.h

```diff
@@ -126,7 +127,8 @@ struct CAFFE2_API SparseTensorImpl : public TensorImpl {
         "shrinking the size of dense dimensions (from ", dense_size_original, " to ", dense_size_new, ") on a non-empty sparse tensor is not supported.\n", alt_options_msg);
     }
 
-    if ((!size.equals(sizes_)) || (sparse_dim != sparse_dim_) || (dense_dim != dense_dim_)) {
+    const bool size_equals_sizes = size.size() == sizes_and_strides_.size() && memcmp(size.data(), sizes_and_strides_.sizes_data(), sizes_and_strides_.size() * sizeof(*size.data())) == 0;
```

Comment:
Surely there is some helper method we could add here :)

Reply:
`std::equal`? :)